eralda/data/components.py

```diff
@@ -0,0 +1,82 @@
+import einops
+import numpy as np
+from pydantic import BaseModel
+
+from ..utils.typing import Array
+
+
+class Pose(BaseModel):
+    """Pose object modelling a set of poses in 3D
+
+    Attributes
+    ----------
+    positions : (n, 3, 1) np.ndarray
+        Positions in 3D represented as column vectors
```

Comment:
I find colum vectors to be less intuitive to work with... why not convert internally?

Reply:
this object actually coerces the input into `(n, 3, 1)` based on the type annotation, you can pass anything which can be coerced into `(n, 3, 1)` e.g. `(n, 3)` and it will work perfectly, does that seem okay?

Reply:
I forgot that this is pydantic's stuff! Cool then :)